### DISCUSSION TITLE
Use configurable optional route segment to detect current locale

### DISCUSSION THREAD

**Discussion by akamfoad** [sergiodxa/remix-i18next#108] [Ideas]
[Remix v1.9.0](https://github.com/remix-run/remix/releases/tag/remix%401.9.0) has been released on top of [React Router v6.5.0](https://github.com/remix-run/react-router/releases/tag/react-router%406.5.0) which adds a new feature called **optional routes**. Optional routes can be defined as `(segment)` where `segment` is any normal RR route, that makes the segment optional, its most obvious advantage could be i18n. It's a common practice to put the locale in the URL which makes it easy and obvious for both users and devs to figure out what is the current locale.

For example, these file structure:

```
- ($lang)/
  - index.jsx
  - blog.jsx
```

Will match
- `/`
- `/blog`
- `/en`
- `/en/blog`
- `/ku`
- `/ku/blog`

We can use this URL param as one of the detection orders, one issue though, the language detector is currently is client side only atm, see:
https://github.com/i18next/i18next-browser-languageDetector/blob/57c6b44da8cf4ee372ba28788cda5b1d3a55ca8a/src/browserLookups/path.js

I'd love to see this feature utilized by remix-i18next

**Comment by sergiodxa** [comment]
In remix-i18next you only use i18next-browser-languageDetector to know the language from the `<html lang>` tag, the locale is detected server-side using https://github.com/sergiodxa/remix-i18next/blob/439127ced69fd5302e31a97683be4838b486fd74/src/server.ts#L180

I'm not sure what's the benefit of this, you will have to pass the params to getFixedT or getLocale, if you do that you can also do `params.lang` or `params.locale` or `params.lng` or however you call it and set a default value, then you can do `i18n.getFixedT(params.locale ?? "en")` for example, as you can see in the types of the method https://github.com/sergiodxa/remix-i18next/blob/439127ced69fd5302e31a97683be4838b486fd74/src/server.ts#L129-L133

**Reply by akamfoad** [reply]
I think I over thinked it, instead of relying in cookies or other ways on the server, we can just use the `lang` param, thanks @sergiodxa for the answer.
